### PR TITLE
[ENH] Add type hints and improve docstring for prep_skl_df utility function

### LINKED ISSUE #857: [DOC] Improve type hints and documentation for prep_skl_df utility function
````
**Describe the documentation issue**

The utility function `prep_skl_df` in `skpro/utils/sklearn.py` currently lacks type hints and contains a slightly inaccurate docstring description.

The existing docstring states that the function *"turns column index into a list of strings"*, while the implementation actually ensures that column names are converted to **string type** when necessary. This can be confusing for contributors reading the documentation.

Additionally, the function signature does not include type hints, which makes it less clear for developers using IDE autocomplete or static analysis tools to understand the expected input and return types.

**Suggest a concrete fix/improvement**

Update the function signature to include type hints and correct the docstring so that it accurately reflects the function’s behavior.

Example improvement:

```python
def prep_skl_df(df: pd.DataFrame, copy_df: bool = False) -> pd.DataFrame:
````

The docstring should clarify that the function ensures column names are converted to string type when necessary and include a proper `Returns` section describing the returned DataFrame.

These changes would improve documentation clarity, developer experience, and IDE support without modifying the function’s behavior.

```

### PR DESCRIPTION
**Pull request by MayankSharma-2812**
#### Reference Issues/PRs
Fixes #857

#### What does this implement/fix? Explain your changes.
Enhanced the [prep_skl_df](cci:1://file:///c:/Users/mayan/skpro/skpro/utils/sklearn.py:7:0-31:13) utility function with proper type hints and improved documentation.

**Changes made:**
- Added type hints: `df: pd.DataFrame`, `copy_df: bool = False`, `-> pd.DataFrame`
- Removed unused import: `from typing import Union`
- Improved docstring accuracy: "turns column index into a list of strings" → "Ensures that the column index consists of strings by converting column names to string type if necessary"
- Added proper return type documentation section

#### Does your contribution introduce a new dependency? If yes, which one?
No, this uses existing pandas imports and removes unused typing import.

#### What should a reviewer concentrate their feedback on?
* Verify that type hints are correct and follow Python typing conventions
* Ensure docstring accurately describes the function's behavior
* Test that the function still works correctly with the changes
* Check that no other unused imports exist in the same file

#### Did you add any tests for the change?
No, this is a documentation and type hints improvement that doesn't change functional behavior.

#### Any other comments?
This is a moderate-size improvement that enhances code quality, IDE support, and maintainability. The function is used throughout the codebase for sklearn compatibility, making this improvement high-impact while being safe and easy to review.

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/sktime/skpro/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [ENH] - adding or improving code, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [ ] I've added estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
- [ ] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.
- [ ] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

**Comment by MayankSharma-2812** [comment]
Thanks for the feedback!

You're right that my previous revision removed some useful details about the behavior of `copy_df`. I've updated the docstring to restore that explanation while also removing the "Changes" section and improving the `Returns` description to better follow numpydoc format.

> This actually removes information and clarity from the docstring, e.g., what happens in the `False` case. Not a good idea.
> 
> The "returns" section also does not adds more information about what is returned.
> 
> "changes" is not a numpydoc header.
